vm/src/types/slot.rs

```diff
@@ -414,7 +419,17 @@ impl PyType {
                 update_pointer_slot!(as_mapping, mapping_methods);
             }
             _ if name == identifier!(ctx, __hash__) => {
-                toggle_slot!(hash, hash_wrapper);
+                let is_unhashable = self
+                    .attributes
+                    .read()
+                    .get(identifier!(ctx, __hash__))
+                    .map(|a| a.is(&ctx.none()))
+                    .unwrap_or(false);
```

Comment:
```suggestion
                    .map_or(false, |a| a.is(&ctx.none));
```

`ctx.none()` increase reference count of `None`. We don't need to do.